staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -107,6 +107,37 @@ func proxyError(w http.ResponseWriter, req *http.Request, error string, code int
 }
 
 func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	w = newStatusResponseWriter(w, req)
+	errRsp := newHijackResponder(&responder{w: w}, req)
+	// TODO: to builder pattern
+	retryManager := newHijackProtector(w.(*statusResponseWriter), newMaxRetries(newRetryDetector(errRsp), 3))
+
+	for {
+		// TODO: do we have to clone the req ?
```

Comment:
am pretty sure we have to.

Also, are we sure `w` was not written to?

Reply:
> Also, are we sure w was not written to?

yes as long as `statusCode and wasHijacked` weren't set 

https://github.com/p0lyn0mial/kubernetes/blob/80f8bf511ceeef0dac6293058c0f80023e15710a/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go#L133